Doc/library/collections.abc.rst

```diff
@@ -224,36 +227,71 @@ Collections Abstract Base Classes -- Detailed Descriptions
    The only reliable way to determine whether an object is :term:`iterable`
    is to call ``iter(obj)``.
 
-.. class:: Collection
+.. class:: Collection[T_co](Sized, Iterable[T_co], Container[T_co])
 
    ABC for sized iterable container classes.
 
    .. versionadded:: 3.6
 
-.. class:: Iterator
+.. class:: Iterator[T_co](Iterable[T_co])
 
    ABC for classes that provide the :meth:`~iterator.__iter__` and
    :meth:`~iterator.__next__` methods.  See also the definition of
    :term:`iterator`.
 
-.. class:: Reversible
+.. class:: Reversible[T_co](Iterable[T_co])
 
    ABC for iterable classes that also provide the :meth:`~object.__reversed__`
    method.
 
    .. versionadded:: 3.6
 
-.. class:: Generator
+.. class:: Generator[YieldType_co, SendType_contra, ReturnType_co](Iterator[YieldType_co])
 
    ABC for :term:`generator` classes that implement the protocol defined in
    :pep:`342` that extends :term:`iterators <iterator>` with the
    :meth:`~generator.send`,
    :meth:`~generator.throw` and :meth:`~generator.close` methods.
 
+   A generator can be annotated by the generic type
+   ``Generator[YieldType, SendType, ReturnType]``. For example::
+
+      def echo_round() -> Generator[int, float, str]:
+          sent = yield 0
+          while sent >= 0:
+              sent = yield round(sent)
+          return 'Done'
```

Comment:
Most people won't go to the `collections.abc` docs for detailed information on type-hinting. I agree that this information doesn't belong in the docs for the deprecated `typing.Generator` symbol anymore, but I'm not sure this is the right solution. I'd advocate moving the docs on generators and coroutines to an "Annotating generators and coroutines" section towards the top of the `typing` docs, similar to what I did for tuples in https://github.com/python/cpython/commit/968435ddb1c1af9333befb26f7970cded8a5c710, for callables in https://github.com/python/cpython/commit/ca8b55c7f54b38e264056148075a8061a7082013, and for `typing.Type` in https://github.com/python/cpython/commit/3eeb8c89063d5ac22c0b1d26e4ae2fd12c149650

Reply:
(FWIW, I _do_ think it would be useful to _link_ from the docs for `collections.abc.Generator` to a guide in the typing docs on how to type-hint generators!)